qiskit/tools/jupyter/version_table.py

```diff
@@ -40,11 +40,24 @@ def qiskit_version_table(self, line="", cell=None):
         qver = qiskit.__qiskit_version__
 
         packages.append(("Qiskit", qver["qiskit"]))
-        packages.append(("Terra", qver["qiskit-terra"]))
-        packages.append(("Aer", qver["qiskit-aer"]))
-        packages.append(("Ignis", qver["qiskit-ignis"]))
-        packages.append(("Aqua", qver["qiskit-aqua"]))
-        packages.append(("IBM Q Provider", qver["qiskit-ibmq-provider"]))
+        if qver["qiskit-terra"]:
+            packages.append(("Terra", qver["qiskit-terra"]))
+        if qver["qiskit-aer"]:
+            packages.append(("Aer", qver["qiskit-aer"]))
+        if qver["qiskit-ignis"]:
+            packages.append(("Ignis", qver["qiskit-ignis"]))
+        if qver["qiskit-aqua"]:
+            packages.append(("Aqua", qver["qiskit-aqua"]))
+        if qver["qiskit-optimization"]:
+            packages.append(("Optimization", qver["qiskit-optimization"]))
+        if qver["qiskit-machine-learning"]:
+            packages.append(("Machine Learning", qver["qiskit-machine-learning"]))
+        if qver["qiskit-nature"]:
+            packages.append(("Nature", qver["qiskit-nature"]))
+        if qver["qiskit-finance"]:
+            packages.append(("Finance", qver["qiskit-finance"]))
+        if qver["qiskit-ibmq-provider"]:
+            packages.append(("IBM Q Provider", qver["qiskit-ibmq-provider"]))
```

Comment:
Instead of listing all of these one by one why not just do this as a loop over the version dictionary? With the exception of the ibmq provider the name is trivial to get from the package name in the dict. The reason I'm thinking move to a more programmatic approach is so we don't have to update this constantly in lock step wit the version dict. Like in the near future there will be new projects added to the dict (experiments and eventually dynamics too) so limiting the amount of things to update around this would be good.

Reply:
in that case, could it be better to have "`qiskit-ibmq-provider`" instead of "IBM Q Provider"?

Reply:
I was thinking more you do something like:
```python
for package in qver:
    if 'ibmq' in package:
        packages.append('IBMQ Provider', qver[package])
    else:
        package_component = package.split('qiskit-')[1:]
        out_name = ' '.join([x.capitalize() for x in package_component.split('-')])
        packages.append(out_name, qver[package])
```

Reply:
I think is a good thing to have the literal name of the package, they name of the column is _Qiskit Software_. You dont like it?

Reply:
I'm fine with package name. I just assumed there was a preference to keep the same naming. But if not then yeah just use the package name from the dict it's much easier.

Reply:
done in 93648b1adc39a0b5570234e718b2bfc75b900d4d